xarray/tests/test_units.py

```diff
@@ -2231,6 +2237,21 @@ def test_pad_unit_constant_value(self, unit, error, dtype):
         assert_units_equal(expected, actual)
         assert_identical(expected, actual)
 
+    @requires_dask
+    @pytest.mark.xfail
+    def test_tokenize_duck_dask_array(self, dtype):
+        import dask
+
+        array = dask.array.linspace(0, 5, 3 * 10).reshape(3, 10).astype(dtype)
+        q = unit_registry.Quantity(array, unit_registry.m)
+        variable = xr.Variable(("x", "y"), q)
+
+        token = dask.base.tokenize(variable)
+        post_op = variable + 5 * unit_registry.m
+
+        assert dask.base.tokenize(variable) != dask.base.tokenize(post_op)
+        assert dask.base.tokenize(variable) == token
```

Comment:
this last line seems unnecessary.

Reply:
Yes, it seems redundant, but as I was working on [hgrecco/pint#1151](https://github.com/hgrecco/pint/pull/1151) and testing this I found that if `__dask_tokenize__` is not implemented by the object passed to `dask.base.tokenize` the token for the original object changes after an operation. Many tests in pint have similar immutability checks.

Reply:
I can confirm that the original token does indeed change after an operation is performed if `__dask_tokenize__` is not implemented, so I left the immutability assertion in place.